src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala

```diff
@@ -0,0 +1,69 @@
+package uk.gov.nationalarchives.downloadfiles
+
+import java.net.URLDecoder
+import java.nio.file.Paths
+import java.util.UUID
+
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3EventNotificationRecord
+import com.typesafe.config.{Config, ConfigFactory}
+import graphql.codegen.GetOriginalPath.getOriginalPath.{Data, Variables, document}
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
+import sttp.client.{Identity, NothingT, SttpBackend}
+import uk.gov.nationalarchives.tdr.{GraphQLClient, GraphQlResponse}
+import uk.gov.nationalarchives.tdr.error.NotAuthorisedError
+import uk.gov.nationalarchives.tdr.keycloak.KeycloakUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class FileUtils()(implicit val executionContext: ExecutionContext) {
+  val config: Config = ConfigFactory.load
+
+  def getFilePath(keycloakUtils: KeycloakUtils, client: GraphQLClient[Data, Variables], fileId: UUID)(implicit backend: SttpBackend[Identity, Nothing, NothingT]): Future[Either[String, String]] = {
+
+    val queryResult: Future[Either[String, GraphQlResponse[Data]]] = (for {
+      token <- keycloakUtils.serviceAccountToken(config.getString("auth.client.id"), config.getString("auth.client.secret"))
+      result <- client.getResult(token, document, Option(Variables(fileId)))
+    } yield Right(result)) recover(e => {
+      Left(e.getMessage)
+    })
+
+    val result = queryResult.map {
+      case Right(response) => response.errors match {
+        case Nil => Right(response.data.get)
+        case List(authError: NotAuthorisedError) => Left(authError.message)
```

Comment:
And another one.

We might as well make sure the full error is captured now, because it's the first thing we'll want to do when we hit an error that's hard to debug in production. 🙂